ckanext/recombinant/errors.py

```diff
@@ -8,3 +9,21 @@ class RecombinantConfigurationError(Exception):
 class BadExcelData(Exception):
     def __init__(self, message):
         self.message = message
+
+
+def format_trigger_error(error_values: list):
+    """
+    Format PSQL function errors from raised ValidationError exceptions.
+
+    This method will split the error messages on a unicode private code point (\\uF8FF)
+    in order to do string replacements, allowing i18n support in the framework.
+    """
+    for e in error_values:
+        if '\uF8FF' in e:
+            yield _(e.split('\uF8FF')[0]).format(e.split('\uF8FF')[1])
+        # NOTE: unicode-escape to handle any raw strings from the triggers.
+        elif '\uF8FF' in e.encode().decode('unicode-escape'):
+            err = e.encode().decode('unicode-escape')
```

Comment:
raw strings from the triggers?

Reply:
this encode-decode sequence will break non-ascii characters so it would be better if we could have the actual private code point instead of a backslash-escape in the source string. Where is the string that needs this is coming from?

Reply:
@wardi some of our trigger/functions in triggers.py in Canada plugin return raw strings.

Reply:
raw strings are just a different syntax for literal strings (they're the same type)